modules/rational-defaults.el

```diff
@@ -11,14 +11,44 @@
 
 ;;; Code:
 
+;; Defining variable for XDG directories specifications
+(defvar xdg-data-home-directory
```

Comment:
we have `rational-config-var-directory` and `rational-config-etc-directory`, so not sure we need additional configuration for yet 2 more folders to spread Emacs files over. If we were not going to use the rational directories, these would certainly make sense, and then you probably would want `(expand-file-name ".local/share/" (getenv "HOME"))` so the full path is set rather than the relative one from wherever Emacs is started.

This may be an opportunity to introduce a `rational-use-xdg-folders` variable and then prefer those over the `rational-config-*-directory` folders.